q2_cutadapt/_demux.py

```diff
@@ -85,13 +85,11 @@ def _rename_files(seqs_dir_fmt, per_sample_dir_fmt, barcode_series):
                                '%s.%d.fastq.gz' % (sample_id,
                                                    read_direction))
 
-            # TODO: remove this outer guard when we upgrade to cutadapt 3
-            if os.path.isfile(src):
```

Comment:
@thermokarst, it's unclear to me if this guard should actually be removed.   With cutadapt 3, we still have two failing tests:  `test_rename_files_extra_samples_in_barcode_map` for both SE and PE:

```
FileNotFoundError: [Errno 2] No such file or directory: '/var/folders/4s/2h2htjk90mb25_ltqkm05y_40000gp/T/q2-SingleLanePerSampleSingleEndFastqDirFmt-xmp1kwu6/sample_c.1.fastq.gz' -> '/var/folders/4s/2h2htjk90mb25_ltqkm05y_40000gp/T/q2-SingleLanePerSampleSingleEndFastqDirFmt-xmp1kwu6/sample_c_C_L001_R1_001.fastq.gz'
```

Reply:
Oh yeah, good point - The `isfile` check is effectively performing an INNER JOIN for us. I wasn't considering the fact that the list of barcodes are what are driving this loop. I agree, please put the guard back, and update the comment to say something about this observation.

Reply:
Actually, wasn't this right the first time? The tests that were failing were ones that directly tested the util - I think those tests just weren't valid anymore. cutadapt will generate a file for every barcode seen, so this `if os.path.isfile(src)` should always evaluate to `True`.